### ISSUE TITLE
Back to top arrow?

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/twentyseventeen#230]
As one scrolls down the page it is pretty common to have a back to top arrow the user can click to go back to the top again.
Is this something that should be included in Twenty Seventeen?

**Comment by josephfusco** [comment]
Personally not a fan of those but either way I don't think there is a need since the navigation is fixed.

**Comment by lukecav** [comment]
Nothing against scroll to top functions in themes.

**Comment by melchoyce** [comment]
I designed one and then totally forgot about it. ([You can see it at this bottom of this mockup](https://cloudup.com/iaV-IVJgEbp/f).) 

With fixed navigation, I'm not sure we need it. I don't feel super strongly either way, though.

**Comment by melchoyce** [comment]
For what it's worth, a "back to top" is more important on small screens, where the menu isn't fixed.

**Comment by paaljoachim** [comment]
I think it is a feature that is nice to have included.

**Comment by josephfusco** [comment]
To clarify my previous response, I am not a fan of the ones that are always fixed on the bottom of the screen. I find them distracting and sometimes in the way on smaller screens. @melchoyce good point on smaller screens not having a fixed navigation. I can see a need for it there to avoid the marathon scroll to the top.

**Comment by paaljoachim** [comment]
It looks more professional having a button that slowly comes into view as one gets closer to the bottom of the screen.

Something like this...
Code is on the bottom.
https://wordpress.org/support/topic/adding-back-to-top-button-1/?replies=14

**Comment by davidakennedy** [comment]
@paaljoachim Another good issue – thanks for all the ideas.

I don't have a strong opinion here, but I would be open to it since we have a design for it. I'm up for seeing a pull request. It should not be fixed and follow you around. 😄

**Comment by melchoyce** [comment]
> It should not be fixed and follow you around. 

Agreed — I think it can probably hang out in that spot right above the footer.

**Comment by paaljoachim** [comment]
Is the back to top button/arrow almost in place or should I do some work on it?

**Comment by josephfusco** [comment]
@melchoyce is the back to top button going to use a smooth scroll or be instant?

**Comment by josephfusco** [comment]
@paaljoachim AFIK no one has worked on it yet.

**Comment by paaljoachim** [comment]
I have a test site up here:
http://testing-newdefaulttheme-twentyseventeen.qsandbox.com/

On the front page I see the arrow pointing down. No arrow pointing up. I am looking into it.

**Comment by melchoyce** [comment]
> is the back to top button going to use a smooth scroll or be instant?

Instant. More practical on mobile, IMO.

**Comment by melchoyce** [comment]
@paaljoachim: Looks like your test site is password-protected.

**Comment by paaljoachim** [comment]
It looks like it is by password protected by default.
I changed the code to 000 (zero-zero-zero)
I will also see if I can open it up. Or I might add a new site to a multisite I have...

Edit:
Working locally now and will upload to another test site....

**Comment by paaljoachim** [comment]
Sorry to say but I have been working on replicating the arrow that is already in place. Having it in the footer and when the user clicks it it goes up to #primary but have not managed to get it right. http://fragmentsofawalk.com/

So someone should take this over that have better skills then what I have.

**Comment by paaljoachim** [comment]
I have been testing a few variations but I am not getting it to work and will have to give this to someone else to handle. Sorry about that...

**Comment by celloexpressions** [comment]
I like the idea of a scroll-to-top arrow before or just inside the footer for parity with the header scroll down arrow.

**Comment by davidakennedy** [comment]
Since a pull request didn't materialize for this one, I'm going to close it.
